pkg/ccl/backupccl/backup_destination.go

```diff
@@ -353,14 +388,44 @@ func writeNewLatestFile(
 	ctx context.Context, settings *cluster.Settings, exportStore cloud.ExternalStorage, suffix string,
 ) error {
 	// If the cluster is still running on a mixed version, we want to write
-	// to the base directory as well the progress directory. That way if
-	// an old node resumes a backup, it doesn't have to start over.
+	// to the base directory instead of the metadata/latest directory. That
+	// way an old node can still find the LATEST file.
 	if !settings.Version.IsActive(ctx, clusterversion.BackupDoesNotOverwriteLatestAndCheckpoint) {
-		err := cloud.WriteFile(ctx, exportStore, latestFileName, strings.NewReader(suffix))
+		return cloud.WriteFile(ctx, exportStore, latestFileName, strings.NewReader(suffix))
+	}
+
+	// HTTP storage does not support listing and so we cannot rely on the
+	// above-mentioned List method to return us the most recent latest file.
+	// Instead, we will write a latest file once with a well-known filename,
+	// and teach the job to always reach for that filename in the face of
+	// a resume. We may lose progress, but this is a cost we are willing
+	// to pay to uphold write-once semantics.
+	if exportStore.Conf().Provider == roachpb.ExternalStorageProvider_http {
+		var r ioctx.ReadCloserCtx
+		var err error
+		if r, err = exportStore.ReadFile(ctx, latestHistoryDirectory+"/"+latestFileName); err != nil {
+			// TODO (darryl): We should do this only for file not found or directory
+			// does not exist errors. As of right now we only specifically wrap
+			// ReadFile errors for file not found so this is not possible yet.
+			// Since we did not find the latest file this is the first time
+			// we are going to write a latest file, so write it with the well
+			// known filename.
+			return cloud.WriteFile(ctx, exportStore, latestFileName, strings.NewReader(suffix))
+		}
+		err = r.Close(ctx)
 		if err != nil {
 			return err
 		}
 	}
 
-	return cloud.WriteFile(ctx, exportStore, latestHistoryDirectory+"/"+latestFileName, strings.NewReader(suffix))
+	// We timestamp the latest files in order to enforce write once backups.
+	// When the job goes to read these timestamped files, it will List
+	// the latest files and pick the file whose name is lexicographically
+	// sorted to the top. This will be the last latest file we write. It
+	// Takes the one's complement of the timestamp so that files are sorted
+	// lexicographically such that the most recent is always the top.
+	var buffer []byte
```

Comment:
can we use the timestamped helper or move this encoding to a helper similar to the checkpoint file.